pygmt/src/grdcontour.py

```diff
@@ -58,11 +58,11 @@ def grdcontour(self, grid, **kwargs):
         contours specified in ``interval``.
 
         - Specify a fixed annotation interval *annot_int* or a
-          single annotation level +\ *annot_int*.
+          single annotation level **+**\ *annot_int*.
         - Disable all annotation with  **-**.
         - Optional label modifiers can be specified as a single string
-          ``"[annot_int]+e"``  or with a list of arguments
-          ``([annot_int], "e", "f10p", "gred")``.
+          ``"annot_int+e+f10p+gred"`` or with a list of arguments
+          ``[annot_int, "e", "f10p", "gred"]``.
```

Comment:
I am wondering whether here the squared brackets are used to indicate that `annot_int` should be replaced with the desired value by the user. I find this a bit misleading.

Reply:
I don't think the description is correct. In the PyGMT world, a list of arguments means the option is repeatable. For example `frame=["WSen", "xaf", "yaf"]` is translated to `-BWSen -Bxaf -Byaf`. For this parameter, it must be `annot_int+e+f10+gred`, not a list.

Reply:
Hm. I tried the following code to understand the input expected by `annotation`. It works for passing a list and I get the same output as for passing a string.

``` python
# orientated on https://www.pygmt.org/dev/api/generated/pygmt.Figure.grdcontour.html
# last access: 2024/03/18

import pygmt

region = [-92.5, -82.5, -3, 7]

grid = pygmt.datasets.load_earth_relief(resolution="15m", region=region)

fig = pygmt.Figure()
fig.grdcontour(
    projection="M10c",
    region=region,
    grid=grid,
    # annotation="500+f10p+gred",  # string -> WORKS
    # annotation=([500], "f10p", "gred"),  # -> FAILS
    # annotation=(500, "f10p", "gred"),  # -> WORKS
    annotation=[500, "f10p", "gred"],  # list -> WORKS
    frame=True,
)
fig.show()

# fig.savefig(fname="grdcontour_A_string.png")
# fig.savefig(fname="grdcontour_A_list.png")
```

_Output figure_
![grdcontour_A_list](https://github.com/GenericMappingTools/pygmt/assets/94163266/82c8ccf5-7219-46ba-831e-4aa5f2de9450)

Reply:
https://github.com/GenericMappingTools/pygmt/blob/6fd4a00c29a6fe9de9b541823190697a6ec751f7/pygmt/src/grdcontour.py#L30-L32

I see. It's because `A` is defined like above, so the list of arguments is joined by `+`. 

Actually, this is the only parameter that sets to `sequence_plus`.

Reply:
I think we should disallow the syntax like `annotation=[500, "f10p", "gred"]`, because:

1. From a user's perspective, it's no different from passing a string like `500+f10p+gred`
2. The syntax is not well documented and is only used for this single parameter
3. It's not Pythonic at all, and it's very likely that we won't support such syntax in the new alias system (#2949 but still WIP)

So, I think we should remove `A="sequence_plus"` and update the documentation.

Reply:
I was actually also a bit wondering if we need this possibility, as such a list does not improve the readability significantly compared to the corresponding string. So, I think I am fine with removing `A"sequence_plus"`.